src/signature-schemes/BlocklockSignatureScheme.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8;
+
+import {BLS} from "../libraries/BLS.sol";
+
+import {SignatureSchemeBase} from "./SignatureSchemeBase.sol";
+
+/// @title BlocklockSignatureScheme contract
+/// @author Randamu
+/// @dev This contract implements the BLS (Boneh-Lynn-Shacham) signature scheme using the
+/// BN254 curve for the Blocklock protocol.
+/// @dev The contract provides functionality to verify signatures, hash messages to points,
+/// and marshal/unmarshal points for signature verification.
+contract BlocklockSignatureScheme is SignatureSchemeBase {
+    using BLS for bytes;
+
+    string public constant SCHEME_ID = "BN254-BLS-BLOCKLOCK";
+    bytes public constant DST = bytes("BLOCKLOCK_BN254G1_XMD:KECCAK-256_SVDW_RO_H1_");
```

Comment:
would suggest adding the chainId in here, though check with @azixus. Need to bear in mind that we may need to use crosschain DSTs far in the future (ie. verifying a signature for chain A in a smart contract on chain B)

Reply:
Sure, we can do the same as we did for randomness-solidity. I've made a note to add it in today @CluEleSsUK